packages/devtools_app/lib/src/shared/ui/drop_down_button.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+
+import '../analytics/analytics.dart' as ga;
+import '../common_widgets.dart';
+import '../theme.dart';
+
+/// A [DropDownButton] implementation that reports selection changes to our
+/// analytics.
+class AnalyticsDropDownButton<T> extends StatelessWidget {
+  const AnalyticsDropDownButton({
+    super.key,
+    required this.gaScreen,
+    required this.gaDropDownId,
+    required this.message,
+    required this.value,
+    required this.items,
+    required this.onChanged,
+    this.sendAnalytics = true,
+    this.isDense = false,
+  });
+
+  /// The GA ID for the screen this widget is displayed on.
+  final String gaScreen;
+
+  /// The GA ID for this widget.
+  final String gaDropDownId;
+
+  /// Whether to send analytics events to GA.
+  ///
+  /// Only set this to false if [AnalyticsDropDownButton] is being used for
+  /// experimental code we do not want to send GA events for yet.
+  final bool sendAnalytics;
+
+  /// The message to be displayed in the widget's tooltip.
+  final String? message;
+
+  /// The currently selected value.
+  final T? value;
+
+  /// The list of options available in the drop down with their associated GA
+  /// IDs.
+  final List<({DropdownMenuItem<T> item, String gaId})>? items;
+
+  /// Invoked when the selected drop down item has changed.
+  final void Function(T?)? onChanged;
+
+  final bool isDense;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: defaultButtonHeight,
+      child: DevToolsTooltip(
+        message: message,
+        child: RoundedDropDownButton<T>(
+          isDense: isDense,
+          style: Theme.of(context).textTheme.bodyMedium,
+          value: value,
+          items: items?.map((e) => e.item).toList(),
+          onChanged: _onChanged,
+        ),
+      ),
+    );
+  }
+
+  void _onChanged(T? newValue) {
+    if (sendAnalytics) {
+      final gaId = items
+              ?.firstWhereOrNull(
+                (element) => element.item == newValue,
+              )
+              ?.gaId ??
+          '<empty>';
```

Comment:
do we want to assert that gaId is not null? I don't think we should ever hit this case, and then if we did for some reason, would we still want to send analytics for `<empty>`?

Reply:
`gaId` will never be null, but I think `newValue` could be if the selection is cleared. Is it possible for a drop down button selection to be cleared? If not, I think we can do a null assertion here and drop the `<empty>` case.

Reply:
I'm going to assume that we can't clear the selection from the standard UI unless we explicitly set `value` to `null`. In that case, we'll just throw a `StateError` when `list.firstWhere(...)` fails to find a match.